### PR TITLE
Provide: offset query param

### PR DESCRIPTION
**Pull request by omani**
as requested.

### CODE REVIEW

File: mem.go [hunk 1]
```diff
@@ -209,7 +212,7 @@ func (m *MemoryHandler) Find(ctx context.Context, lookup *resource.Lookup, page,
 				end = total
 			}
 		}
-		list = &resource.ItemList{total, page, items[start:end]}
+		list = &resource.ItemList{total, page, offset, items[start:end]}
```
Comment (rs) [inline mem.go]: We should name the fields here.

**Comment by omani** [comment]
done.

**Comment by omani** [comment]
updated.

File: mem.go [hunk 2]
```diff
@@ -198,18 +198,19 @@ func (m *MemoryHandler) Find(ctx context.Context, lookup *resource.Lookup, page,
...
+			end = start + limit
 			if start > total-1 {
 				start = 0
 				end = 0
 			} else if end > total-1 {
 				end = total
 			}
 		}
-		list = &resource.ItemList{total, page, items[start:end]}
+		list = &resource.ItemList{Total: total, Page: offset, Offset: offset, Items: items[start:end]}
```
Comment (rs) [inline mem.go]: That's an issue, here: you can no longer populate the `Page` field as it is not abstracted to the storage handler. I guess it should be rest-layer's responsibility to fill those fields now.

I don't think we should store the `Offset` field in `ItemList`. As mentioned on rest-layer's side, I think we should call that `Skip` in order to not confuse `offset` and `skip` behavior.

Comment (omani) [reply mem.go]: for this I would need to introduce a new ItemList field named Skip. Did I understand correctly?

Comment (rs) [reply mem.go]: I would rename `Offset` to `Skip` and use `skip` query-string parameter instead of `offset`.

File: mem.go [hunk 3]
```diff
@@ -198,18 +198,19 @@ func (m *MemoryHandler) Find(ctx context.Context, lookup *resource.Lookup, page,
...
+			end = start + limit
 			if start > total-1 {
 				start = 0
 				end = 0
 			} else if end > total-1 {
 				end = total
 			}
 		}
-		list = &resource.ItemList{total, page, items[start:end]}
+		list = &resource.ItemList{Total: total, Page: offset, Skip: offset, Items: items[start:end]}
```
Comment (rs) [inline mem.go]: It's still wrong, the `page` is not the `offset` and the `skip` is not the `offset` either.

In this setup, the `offset` is computed from the `page` and `skip` API parameter. This means the storage engine is no longer able to know the value of those parameters, and thus can't init those value in the `ItemList` type. I think those two values (`Page` and `Skip`) should now be set by the caller (rest-layer `resource` package).

Does it make sense?

Comment (omani) [reply mem.go]: Absolutely. I agree it makes things easier.

**Comment by rs** [comment]
I will make the necessary changes and merge.

**Pull request closed without merging**